mongomock/database.py

```diff
@@ -68,10 +72,37 @@ def collection_names(self, include_system_collections=True, session=None):
 
         return self.list_collection_names(session=session)
 
-    def list_collection_names(self, session=None):
+    def _verify_list_collection_supported_op(self, keys):
+        if set(keys) - list_collection_filter_allowed_operators:
+            raise NotImplementedError(
+                'list collection names filter operator {0} is not implemented yet in mongomock '
+                'allowed operators are {1}'.format(keys, list_collection_filter_allowed_operators))
+
+    def list_collection_names(self, filter=None, session=None):
+        """filter: only name field type with eq,ne or regex operator
+
+        session: not supported
+        supported operator are $eq, $ne, $regex
+        """
+        field_name = 'name'
+
         if session:
             raise NotImplementedError('Mongomock does not handle sessions yet')
 
+        if filter:
+            if not filter.get('name'):
+                raise NotImplementedError(' list collection {0} might be valid but is not '
```

Comment:
Please remove the white spaces before `list collection` and before `implemented`